### ISSUE TITLE
Add dashboard shortcut to bottom drawer

### ISSUE THREAD

**Issue by mahalakshme** [avniproject/avni-client#1328]
### Need:
Users frequently dont use search to navigate the beneficiaries. They use the dashboards. Hence having one another dashboard as bottom drawer shortcut helps in easy navigation.

### Acceptance criteria:
- In the Dashboard tab of a user group in webapp, on the right of 'Is Primary' column, add a new column called 'Is Secondary' where one of the dashboards can be marked as Secondary
- The dashboard marked as Secondary need to display before the More icon. Adjust the styling and spacing such that all icons are of same size, etc.,
- Use the dashboard icon under Material Icons [here](https://oblador.github.io/react-native-vector-icons/) as the icon to indicate the dashboard shortcut.
- The name of the dashboard shortcut should be the name of the dashboard. Show only first 7 characters with epsilon following for longer names(names greater than 10 characters).
- Clicking on it in the mobile app, need to take to the dashboard
- Make sure this secondary dashboard configuration is available in the exported bundle and hence in the org to which the bundle is imported to.
- Do not load the secondary dashboard report cards until the icon in the bottom drawer is clicked.

### Out of scope:
- Adding a short name for the dashboard to show in the bottom bar is out of scope. Currently the name used for the dashboard in focus, doesn't convey the meaning and hence need to be changed. Like in the mockup if it is changed to 'Scorecard' would not probably need to configure a short name. Can be decided after the showcase.
- Resolving the issue when this configuration of secondary dashboard is present in multiple user groups to which a user belongs to. 

### Mockups
![dashboard_bottom_drawer](https://github.com/avniproject/avni-client/assets/10369963/fb1cead9-12c0-4d12-89c8-fa88781cb7ad)

### One of the test cases:
Being able to add 'Overall Overview' dashboard as shortcut in the bottom drawer for all user groups in TeachAP app.

**Comment by petmongrels** [comment]
### other commits
- https://github.com/avniproject/avni-server/commit/05fef587eef68cb4927829e5b7111b9b065a9630
- https://github.com/avniproject/avni-server/commit/6f6b1e46fde040c4020bdd6f2e426b23b857848b
- https://github.com/avniproject/avni-server/commit/0156e9ab9dedde0e1e0e6d4c87ba7d4d59124524
- https://github.com/avniproject/avni-webapp/commit/a89c172b60fa0b03541c0cdcc841e364fe7a4c2b

**Comment by petmongrels** [comment]
- allowing user to make the same dashboard primary as well as secondary. allow user to unselect a selected radio - so they can fix it
- user can set secondary dashboard without choosing primary

**Comment by Dinesh2019** [comment]
- [ ] Can we short the name of the dashboard 

<img width="489" alt="Screenshot 2024-03-27 at 2 40 05 PM" src="https://github.com/avniproject/avni-client/assets/127730126/2c3b9972-ef7e-4c74-bdf6-044bc000df02">

- [ ] If we haven't chosen either the primary or secondary options, could we include the dashboard within the "More" icon on the dashboard

<img width="1710" alt="Screenshot 2024-03-27 at 2 45 24 PM" src="https://github.com/avniproject/avni-client/assets/127730126/ae1d3d67-9f66-4459-ab7a-e2b500233016">

**Comment by petmongrels** [comment]
- made dashboard name upto 14 chars
- left second one as it seems like new requirement and doesn't seem must have

**Comment by Dinesh2019** [comment]
- [ ] Still we have move char in the secondary dashboard



![Image](https://github.com/avniproject/avni-client/assets/127730126/a851cb70-27e3-42d2-aed7-63826ecf4519)

**Comment by Dinesh2019** [comment]
Tested env : pre-release 
apk used for testing : [apk](https://app.circleci.com/pipelines/github/avniproject/avni-client/2431/workflows/889620dc-3dce-4701-8599-f494041c8b7e/jobs/15311/artifacts)

 [x] Display dashboard 1 as primary and dashboard 2 as secondary on the home page with respective cards. Dashboard 2 should be accompanied by an icon and a name.
- [x] Limit the name of each dashboard to 14 characters. If the name exceeds 14 characters, trim it to 14 characters before displaying.
- [x] Allow access to a single dashboard as both primary and secondary; both should be visible.
- [x] Grant access to dashboard 1 and dashboard 2 as primary, but ensure users can select only one as the primary dashboard.
- [x] Grant access to dashboard 1 and dashboard 2 as secondary, but ensure users can select only one as the secondary dashboard.
- [x] Create three dashboards and designate one as primary. Attempting to designate more than one dashboard as secondary should be restricted.
- [x] Create three dashboards and designate one as secondary. Attempting to designate more than one dashboard as primary should be restricted.
- [x] Removing primary or secondary permissions from a dashboard should revoke access without automatically assigning permissions to other dashboards.
- [x]  if we didn't gave the permission to any of the dashboard then it will not display in the dashboard option and inbuilt dashboard should be displayed in the home page. can you rephrase it

**Comment by Dinesh2019** [comment]
We unable to see the secondary option in the user group to add the secondary dashboard.Hence reopening the ticket 

env : pre-release
org : iphscd
user : vedantr@iphscd

Refer image :




![Image](https://github.com/avniproject/avni-client/assets/127730126/cc2af3b0-d406-4008-bb57-aae777aa87d9)

**Comment by petmongrels** [comment]
web app for 8.0 not deployed on prerelease

**Comment by Dinesh2019** [comment]
But i have tested this is in per-release already u can see my comment which was added in mar 27

**Comment by petmongrels** [comment]
web app was deployed in 2 months ago in prerelease. in two months likely some other branch was deployed. I can see that the code is not correct.

**Comment by Dinesh2019** [comment]
@petmongrels do we have workaround for this because the card is in QA Ready

**Comment by petmongrels** [comment]
redeploy web app

**Comment by Dinesh2019** [comment]
when we click on the more option the second dashboard is also getting selected but it display on the more option details .

Refer video :



https://github.com/avniproject/avni-client/assets/127730126/3eaf6e02-bdc4-4c9b-9898-efeea33f990b

**Comment by petmongrels** [comment]
@mahalakshme do we want to fix this. I think showing all dashboards doesn't hurt.

**Comment by mahalakshme** [comment]
@petmongrels discussed with Dinesh, here showing all dashboards is not the issue. The issue is just that, when More option is selected, the secondary dashboard option is also getting highlighted in the bottom drawer and it need not. So moving it back to QA failed. Let me know if you think otherwise.
